valkyrie/lib/valkyrie/storage/fedora.rb

```diff
@@ -16,8 +16,12 @@ def handles?(id:)
     # Return the file associated with the given identifier
     # @param id [Valkyrie::ID]
     # @return [Valkyrie::StorageAdapter::StreamFile]
+    # @return [nil] if nothing is found
     def find_by(id:)
       Valkyrie::StorageAdapter::StreamFile.new(id: id, io: response(id: id))
+    rescue ::Ldp::Gone
+      Valkyrie.logger.warn "Unable to find Fedora file with id #{id}"
+      nil
```

Comment:
Same issue here? It's really going to depend on how we envision people consuming the method: checking for nil versus catching the exception at the consumer's point.